### DISCUSSION TITLE
ScreenLogic vs RS485?

### DISCUSSION THREAD

**Discussion by NathanSweet** [parnic/node-screenlogic#60] [Q&A] [answered]
The Pentair ScreenLogic2 "protocol adapter" box has 4 wires and connects to the LAN via ethernet.

I've come across [tagyoureit/nodejs-poolController](https://github.com/tagyoureit/nodejs-poolController) which I guess connects to the same 4 wires the ScreenLogic does. Can everything done via ScreenLogic also be done via RS485?

**Comment by parnic** [comment, marked as answer]
I unfortunately don't know. This library's intent is to interact with the remote TCP API. Presumably you'd be able to do the same stuff over a wire, but that repo you mentioned is your best bet for that. I'm not much of a hardware guy.

**Reply by NathanSweet** [reply]
Cool, no problem. I'm not familiar with owning a pool/spa, so I'm just trying to find my way. So far I just want control for turning on/off the lights and spa. For the spa, I guess that means both pump and heater.

Do you find the ScreenLogic UI helpful, or do you mainly just use if for control via your library?

I'm trying to decide between these:
1) ScreenLogic + your library. This seems pretty straightforward and closest to the manufacturer's way of doing things, but I can't tell if I'm missing any functionality I may want later.
2) An RPi connected via RS485 (no ScreenLogic). The library I linked seems pretty extensive. FWIW, I don't benefit from any of the "Home Automation Bindings", since I roll my own automation software.
3) ScreenLogic + RS485. I doubt I'd do both, since if I get ScreenLogic it seems easier to talk to.

**Reply by parnic** [reply]
I have been implementing each new message as we discover them and have need. I use the official app when I need to change stuff on my pool while we're outside using it, and this library via https://github.com/parnic/MMM-ScreenLogic for monitoring and while inside the house. Use whatever fits your preferred use case. :)
